src/main.ts

```diff
@@ -65,15 +57,26 @@ export async function main(){
         - voulme mount .azure session token file between host and container,
         - volume mount temp directory between host and container, inline script file is created in temp directory
         */
-        let command: string = `run --workdir ${process.env.GITHUB_WORKSPACE} -v ${process.env.GITHUB_WORKSPACE}:${process.env.GITHUB_WORKSPACE} `;
-        command += ` -v ${process.env.HOME}/.azure:/root/.azure -v ${TEMP_DIRECTORY}:${TEMP_DIRECTORY} `;
-        command += ` ${environmentVariables} `;
-        command += `--name ${CONTAINER_NAME} `;
-        command += ` mcr.microsoft.com/azure-cli:${azcliversion} ${startCommand}`;
+        let command: string = `run`;
+        var args = ["--workdir", `${process.env.GITHUB_WORKSPACE}`,
+            "-v", `${process.env.GITHUB_WORKSPACE}:${process.env.GITHUB_WORKSPACE}`,
+            "-v", `${process.env.HOME}/.azure:/root/.azure`,
+            "-v", `${TEMP_DIRECTORY}:${TEMP_DIRECTORY}`
+        ];
+        for (let key in process.env) {
+            if (!checkIfEnvironmentVariableIsOmitted(key) && process.env[key]) {
+                args = args.concat("-e", `${key}=${process.env[key]}`);
+            }
+        }
+        args = args.concat("--name", `${CONTAINER_NAME}`,
+            `mcr.microsoft.com/azure-cli:${azcliversion}`,
+            `bash`, `--noprofile`, `--norc`, `-e`, `${TEMP_DIRECTORY}/${scriptFileName}`);

```

Comment:
[Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) is not needed for plain text string.